File: app/src/main/java/com/summer/itis/summerproject/model/Card.java
package com.summer.itis.summerproject.model;

import com.summer.itis.summerproject.model.AbstractCard;

public class Card {

    private String id;

    private String cardId;

    private AbstractCard abstractCard;

    private Integer intelligence;

    private Integer support;

    private Integer prestige;

    private Integer hp;

    private Integer strength;

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getCardId() {
        return cardId;
    }

    public void setCardId(String cardId) {
        this.cardId = cardId;
    }

    public AbstractCard getAbstractCard() {
        return abstractCard;
    }

    public void setAbstractCard(AbstractCard abstractCard) {
        this.abstractCard = abstractCard;
    }

    public Integer getIntelligence() {
        return intelligence;
    }

    public void setIntelligence(Integer intelligence) {
        this.intelligence = intelligence;
    }

    public Integer getSupport() {
        return support;
    }

    public void setSupport(Integer support) {
        this.support = support;
    }

    public Integer getPrestige() {
        return prestige;
    }

    public void setPrestige(Integer prestige) {
        this.prestige = prestige;
    }

    public Integer getHp() {
        return hp;
    }

    public void setHp(Integer hp) {
        this.hp = hp;
    }

    public Integer getStrength() {
        return strength;
    }

    public void setStrength(Integer strength) {
        this.strength = strength;
    }
}

File: app/src/main/java/com/summer/itis/summerproject/model/Card.kt
package com.summer.itis.summerproject.model

import com.google.firebase.database.Exclude
import com.google.firebase.database.IgnoreExtraProperties
import com.summer.itis.summerproject.model.AbstractCard

@IgnoreExtraProperties
class Card {

    var id: String? = null

    var cardId: String? = null

    var testId: String? = null

    var intelligence: Int? = null

    var support: Int? = null

    var prestige: Int? = null

    var hp: Int? = null

    var strength: Int? = null

    @Exclude
    var abstractCard: AbstractCard? = AbstractCard()

    @Exclude
    var test: Test = Test()
}
